tl/controllers/middleware.rst

```diff
@@ -129,121 +129,120 @@ scripts, that add middleware::
+    // Ito ay *hindi* nagbabago sa $response. Ang bagong object ay hindi
+    // itinalaga sa isang variable.
     $response->withHeader('Content-Type', 'application/json');
 
-    // This works!
+    // Ito ay gumagana!
     $newResponse = $response->withHeader('Content-Type', 'application/json');
 
-Most often you'll be setting headers and response bodies on requests::
+Madalas ikaw ay magtatakda ng mga header at mga tugon na kumakatawan sa mga hiling::
 
-    // Assign headers and a status code
+    // Magtakda ng mga header at isang code ng katayuan
     $response = $response->withHeader('Content-Type', 'application/json')
         ->withHeader('Pragma', 'no-cache')
         ->withStatus(422);
 
-    // Write to the body
+    // Sumulat sa katawan
     $body = $response->getBody();
     $body->write(json_encode(['errno' => $errorCode]));
 
-Creating Middleware
+Paglikha ng Middleware
 ===================
 
-Middleware can either be implemented as anonymous functions (Closures), or as
-invokable classes. While Closures are suitable for smaller tasks they make
-testing harder, and can create a complicated ``Application`` class. Middleware
-classes in CakePHP have a few conventions:
+Ang Middleware ay maaaring ipatupad bilang hindi kilalang mga function (Closures), o bilang
+nanawagan na mga class. Habang ang Closures ay karapatdapat para sa maliit na mga gawain na ginagawa nila
+na pag-test sa mas mahirap, at maaaring lumikha ng isang kumplikadong ``Application`` na class. Ang Middleware
+na class sa CakePHP ay nagkaroon ng ilang mga kombensiyon:
 
-* Middleware class files should be put in **src/Middleware**. For example:
+* Ang Middleware class na mga file ay dapat ilagay sa **src/Middleware**. Halimbawa:
   **src/Middleware/CorsMiddleware.php**
-* Middleware classes should be suffixed with ``Middleware``. For example:
+* Ang Middleware na mga class ay dapat naka-suffix sa ``Middleware``. Halimbawa:
   ``LinkMiddleware``.
-* Middleware are expected to implement the middleware protocol.
+* Ang Middleware ay inaasahan na ipatupad ang middleware na protocol.
 
-While not a formal interface (yet), Middleware do have a soft-interface or
-'protocol'. The protocol is as follows:
+Habang hindi sa isang pormal na interface (hindi pa), Ang Middleware ay mayroong isang soft-interface o
+'protocol'. Ang protocol ay ang mga sumusunod:
 
-#. Middleware must implement ``__invoke($request, $response, $next)``.
-#. Middleware must return an object implementing the PSR-7 ``ResponseInterface``.
+#. Ang Middleware ay dapat nagpatupad ng ``__invoke($request, $response, $next)``.
+#. Ang Middleware ay dapat magbalik ng isang object na nagpapatupad sa PSR-7 ``ResponseInterface``.
 
-Middleware can return a response either by calling ``$next`` or by creating
-their own response. We can see both options in our simple middleware::
+Ang Middleware ay maaaring bumalik ng sagot na alinman sa pamamgitan ng pagtawag sa ``$next`` o sa pamamagitan ng paglilikha
```

Comment:
maaaring magsauli ng sagot
(or you could use "magbalik")

Reply:
done change